.github/workflows/PR.yml

```diff
@@ -26,7 +26,7 @@ jobs:
             uses: actions/setup-java@v3
             with:
                distribution: 'temurin'
-               java-version: '8'
+               java-version: '17'
```

Comment:
The JDK used to run the build is no longer determining the target artifacts bytecode, so it's safe to use a more recent JDK.

This could easily be `'20'` as well, but I decided to use an LTS version until I get feedback that more recent versions are OK too. Fwiw I've been running JDK20 for a long time without issues (and JDK19 and 18 before that). Personally I'd suggest to always use the newest JDK for both CI and for the toolchains

Reply:
I think staying on LTS versions makes sense for this project. I dont think we would bother to bump these versions every 6 months, so it's likely we would end up on an unsupported version. Besides, I don't think we gain much from being on latest JDK?

Reply:
I agree. Compatibility is more important than anything else so lets stick to the LTS 17. We can go to 21 when that's out.

Reply:
> Besides, I don't think we gain much from being on latest JDK?

Generally, you do — newer JDKs bring performance improvements and bug fixes. Some are really noticeable (e.g. a metaspace leak that was transparently handled by Gradle but incurred massive memory cost, see https://youtrack.jetbrains.com/issue/KT-55831) but there are also smaller performance fixes, improvements to GC algorithms etc. This may not matter for CI, but this is related to:

> Compatibility is more important than anything else so lets stick to the LTS 17

This PR is specifically fixing incompatibility with JDK20, which is stable and is a valid JDK to run builds with, but Kotest build was failing. So building with JDK17 on CI is not guaranteeing further compatibility with JDK20. It's of course valid to say that Kotest is only _compatible_ with JDK17, but then you would force your consumers to use an old JDK (note that this PR had to change some _tests_ too, which means the consumers face the same issues). So if anything, it'd make sense to test the build on both LTS and current JDKs.

Same goes for toolchains, using newer JDK gives performance improvements for the cost of updating a number in couple of places every half a year. Of course that's your decision (I'm just happy I can now have deterministic output regardless of JDK I use to run the build). I just wanted to point out new JDKs _do_ have benefits and using LTS _doesn't_ solve any compatibility issues in a world where developers use non-LTS releases.